okhttp-client/src/main/scala/org/http4s/client/okhttp/OkHttp.scala

```diff
@@ -0,0 +1,140 @@
+package org.http4s.client.okhttp
+
+import java.io.IOException
+
+import cats.data._
+import cats.effect._
+import cats.effect.implicits._
+import okhttp3.{
+  Call,
+  Callback,
+  OkHttpClient,
+  Protocol,
+  RequestBody,
+  Headers => OKHeaders,
+  MediaType => OKMediaType,
+  Request => OKRequest,
+  Response => OKResponse
+}
+import okio.BufferedSink
+import org.http4s.{Header, Headers, HttpVersion, Method, Request, Response, Status}
+import org.http4s.client.{Client, DisposableResponse}
+import fs2.Stream._
+import fs2._
+import fs2.io._
+
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
+
+object OkHttp {
+
+  def defaultconfig[F[_]]()(implicit F: Effect[F]): F[OkHttpClient.Builder] = F.delay {
```

Comment:
A `Sync` constraint is sufficient to get `delay`.  But is this even a side effect?  Why isn't this pure?

Nit: camelcase `defaultConfig`

Reply:
Creating the `Builder` creates all of the internals which the `OkHttpClient` is going to use, including an `Executor` and a `ConnectionPool`, so during testing when I shared the same configuration between two clients, shutting one down shut down the other.

Reply:
Oh, wow.  I thought that would be deferred until `build()`.  Then I think you're right.